src/collective/cover/tiles/carousel.py

```diff
@@ -66,12 +66,21 @@ def populate_with_object(self, obj):
         :type uuids: List of strings
         """
         super(ListTile, self).populate_with_object(obj)  # check permission
-        uuids = ICoverUIDsProvider(obj).getUIDs()
+        if obj.portal_type in ('Topic', 'Collection'):
+            uuids = [i.UID for i in obj.queryCatalog()]
+        else:
+            uuids = [self.get_uuid(obj)]
         # accept just elements with a lead image
         uuids = [i for i in uuids if self._has_image_field(uuidToObject(i))]
         if uuids:
             self.populate_with_uuids(uuids)
 
+    @view.memoize
+    def accepted_ct(self):
```

Comment:
> the validation must be done in the populate_with_object() method.

How should a message be shown to the user if there's an invalid `portal_type` inside a container being validated by `populate_with_object`? Maybe this approach can be used in brasil.gov.tiles as well.